### PR TITLE
Update observe, rollback cmds

### PR DESCRIPTION
**Pull request by juanvallejo**
Updates `observe` and `rollback` commands with upstream patterns

cc @deads2k

### CODE REVIEW

**Comment by juanvallejo** [comment]
cc @deads2k

File: pkg/oc/cli/rollback/rollback.go [hunk 1]
```diff
@@ -267,22 +267,13 @@ func (o *RollbackOptions) Run() error {
 		return err
 	}
 
-	// If this is a dry run, print and exit.
-	if o.DryRun {
-		describer := describe.NewDeploymentConfigDescriber(o.appsClient, o.kc, newConfig)
-		description, err := describer.Describe(newConfig.Namespace, newConfig.Name, kprinters.DescriberSettings{})
+	// If an output format is specified or this is a DryRun, print and exit.
```
Comment (deads2k) [inline pkg/oc/cli/rollback/rollback.go]: @juanvallejo @soltysh @liggitt @smarterclayton this behavior stands out as big difference to the CLI feel between upstream and us.  I also thing that asking for a printer format does not generally imply that I don't want to perform the actual action.  I think our behavior should change.  Suggestions for a reasonable way to fix it?

**Comment by deads2k** [comment]
/lgtm

could use a squash if you get to it in time.

File: pkg/oc/cli/rollback/rollback.go (hunk 1 above)
Comment (liggitt) [reply pkg/oc/cli/rollback/rollback.go]: other than deprecating the implicit "output means dry run" behavior, starting to print a warning now if they set output without dry-run, and making `--dry-run` required in a future release, not really

Comment (deads2k) [reply pkg/oc/cli/rollback/rollback.go]: > other than deprecating the implicit "output means dry run" behavior, starting to print a warning now if they set output without dry-run, and making --dry-run required in a future release, not really

Long as we thing that's enough, let's do that.

@mfojtik card, bug, or issue?

**Comment by juanvallejo** [comment]
@deads2k squashed. Re-adding lgtm

@mfojtik Added a card to track [David's comment](https://github.com/openshift/origin/pull/20291#discussion_r203699227) here: https://trello.com/c/TFnrZwbx

**Pull request merged**